src/hardhat/utils/index.ts

```diff
@@ -30,69 +37,42 @@ export function getW3fDetails(w3fRootDir: string) {
       } else return;
 
       // get userArgs
-      try {
-        const userArgsJsonString = fs.readFileSync(userArgsJsonPath, "utf8");
-        const userArgs = JSON.parse(userArgsJsonString);
-
-        w3fDetails[file] = {
-          path: indexPath,
-          userArgs,
-          storage: {},
-          secrets: {},
-        };
-      } catch (error) {
-        console.error(
-          `Error reading userArgs.json for ${file}: ${error.message}`
-        );
-        w3fDetails[file] = {
-          path: indexPath,
-          userArgs: {},
-          storage: {},
-          secrets: {},
-        };
-      }
+      if (fs.existsSync(userArgsJsonPath)) {
+        try {
+          const userArgsJsonString = fs.readFileSync(userArgsJsonPath, "utf8");
+          userArgs = JSON.parse(userArgsJsonString);
+        } catch (error) {
+          console.error(
+            `Error reading userArgs.json for ${file}: ${error.message}`
+          );
+        }
+      } else console.warn(`${WARN} userArgs.json not found\n`);
 
       // get storage
-      try {
-        const storageJsonString = fs.readFileSync(storageJsonPath, "utf8");
-        const storage = JSON.parse(storageJsonString);
-
-        w3fDetails[file] = {
-          ...w3fDetails[file],
-          storage,
-        };
-      } catch (error) {
-        console.error(
-          `Error reading storage.json for ${file}: ${error.message}`
-        );
-        w3fDetails[file] = {
-          ...w3fDetails[file],
-          storage: {},
-        };
-      }
+      if (fs.existsSync(storageJsonPath)) {
+        try {
+          const storageJsonString = fs.readFileSync(storageJsonPath, "utf8");
+          storage = JSON.parse(storageJsonString);
+        } catch (error) {
+          console.error(
+            `Error reading storage.json for ${file}: ${error.message}`
+          );
+        }
+      } else console.warn(`${WARN} storage.json not found\n`);
 
       // get secrets
-      try {
-        const secrets: { [key: string]: string } = {};
-        const config = dotenv.config({ path: secretsPath }).parsed ?? {};
-        Object.keys(config).forEach((key) => {
-          secrets[key] = config[key];
-        });
-
-        w3fDetails[file] = {
-          ...w3fDetails[file],
-          secrets,
-        };
-      } catch (error) {
-        console.error(
-          `Error reading storage.json for ${file}: ${error.message}`
-        );
+      if (fs.existsSync(secretsPath)) {
+        try {
+          const config = dotenv.config({ path: secretsPath }).parsed ?? {};
+          Object.keys(config).forEach((key) => {
+            secrets[key] = config[key];
+          });
+        } catch (error) {
+          console.error(`Error reading .env for ${file}: ${error.message}`);
+        }
+      } else console.warn(`${WARN} .env not found\n`);
```

Comment:
same comment, not all web3 functions use secrets

Reply:
[780579f](https://github.com/gelatodigital/web3-functions-sdk/pull/22/commits/780579feb90f9799a7d3d816607ff161cbf6598c)